jbang-catalog.json

```diff
@@ -0,0 +1,123 @@
+{
+  "aliases": {
+    "undertow": {
+      "scriptRef": "undertow.java"
+    },
+    "records": {
+      "scriptRef": "records.java"
+    },
+    "tz": {
+      "scriptRef": "tz.java"
+    },
+    "vertx": {
+      "scriptRef": "vertx.java"
+    },
+    "classpath_example": {
+      "scriptRef": "classpath_example.jsh"
+    },
+    "hibernate": {
+      "scriptRef": "hibernate.java"
+    },
+    "example": {
+      "scriptRef": "example"
+    },
+    "docker": {
+      "scriptRef": "docker.java"
+    },
+    "smee": {
+      "scriptRef": "smee.java"
+    },
+    "analytics": {
+      "scriptRef": "analytics.java"
+    },
+    "update_catalog": {
+      "scriptRef": "update_catalog.java"
+    },
+    "camel": {
+      "scriptRef": "camel.java"
+    },
+    "checksum": {
+      "scriptRef": "checksum.java"
+    },
+    "jfxtiles": {
+      "scriptRef": "jfxtiles.java"
+    },
+    "webdriver": {
+      "scriptRef": "webdriver.java"
+    },
+    "jetty": {
+      "scriptRef": "jetty.jsh"
+    },
+    "scripts": {
+      "scriptRef": "scripts"
+    },
+    "lang": {
+      "scriptRef": "lang.java"
+    },
+    "shrinkwrap": {
+      "scriptRef": "shrinkwrap.java"
+    },
+    "ratpack": {
+      "scriptRef": "ratpack.java"
+    },
+    "asciidoc": {
+      "scriptRef": "asciidoc.java"
+    },
+    "helloworld": {
+      "scriptRef": "helloworld.jsh"
+    },
+    "urldep": {
+      "scriptRef": "urldep.java"
+    },
+    "imap": {
+      "scriptRef": "imap.java"
+    },
+    "sparkjava": {
+      "scriptRef": "sparkjava.java"
+    },
+    "asciidoctor": {
+      "scriptRef": "asciidoctor.java"
+    },
+    "plantuml": {
+      "scriptRef": "plantuml.java"
+    },
+    "inetTest": {
+      "scriptRef": "inetTest.java"
+    },
+    "script": {
+      "scriptRef": "script.java"
+    },
+    "piping": {
+      "scriptRef": "piping.java"
+    },
+    "findmodule": {
+      "scriptRef": "findmodule.java"
+    },
+    "junitrun": {
+      "scriptRef": "junitrun.java"
+    },
+    "resteasyclient": {
+      "scriptRef": "resteasyclient.java"
+    },
+    "progress": {
+      "scriptRef": "progress.java"
+    },
+    "githubinfo": {
+      "scriptRef": "githubinfo.java"
+    },
+    "envdetector": {
+      "scriptRef": "envdetector.java"
+    },
+    "gh_fetch_release_assets": {
+      "scriptRef": "gh_fetch_release_assets.java"
+    },
+    "jfx": {
+      "scriptRef": "jfx.java"
+    },
+    "junit-run": {
+      "scriptRef": "junit-run.java"
+    }
+  },
+  "baseRef": "examples",
```

Comment:
what does this "baseRef" refer to ?

Reply:
It's an optional field that will be applied as a prefix for all the scripts mentioned in the file. So in this case all the scripts are expected to be in the `examples` sub folder. It can also be an absolute URL (in case you want to make a catalog file for scripts that are all in some other repository for example). And in all cases you can also put an absolute URL in the `scriptRef` if the script isn't in the current folder/repo or where the `baseRef` points to.

Reply:
ah ack.

btw. could we use kebab case; i hate json but if i at least don't have to remember casing its tolerable ;)

Reply:
Sure, we can, but that means that current alias files will not be compatible :-/

Reply:
bummer....well...should have caught that earlier on. lets break it now instead of later.

Reply:
should be possible to do a basic string search/replace to fix it if mapping exception occurs and try again.

Reply:
sure, and we should probably do it, but couldn't you have said this _before_ we merged the alias stuff? :-P

Reply:
Thing is that the files get parsed straight from the input streams. So doing this search/replace will make the code unfortunately less nice (first read into a string, search/replace, parse string)